### ISSUE TITLE
Unhandled Edge Case: Style/AccessModifierDeclarations

### ISSUE THREAD

**Issue by CoderMiguel** [rubocop/rubocop#13213]
similar to https://github.com/rubocop/rubocop/issues/5953

## Expected behavior

I expect symbol array literal notation to not be offensive to RuboCop

```ruby
class MyClass
  ... # some public methods
  
  private
  
  def mata_programming_method # to remain private
    # leverages __callee__ (the name of the alias method) for dynamic behavior
    ...
  end
  alias my_first_alias mata_programming_method
  alias my_second_alias mata_programming_method

  public(*%i[my_first_alias my_second_alias]) 
end
```

## Actual behavior

Describe here what actually happened.
Offenses:
```console
[Correctable] Style/AccessModifierDeclarations: public should not be inlined in method definitions.
  public(*%i[my_first_alias my_second_alias])
  ^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```
## Steps to reproduce the problem
refactor a non-offensive use of AllowModifiersOnSymbols: true (default)
```ruby
  public :my_first_alias, :my_second_alias #=> No Offense
```
  to use symbol array literal notation and re-run rubocop
```ruby
  public(*%i[my_first_alias my_second_alias]) # => Style/AccessModifierDeclarations: `public` should not be inlined in method definitions.
```

### Further down the rabbit hole
This is also throwing errors when the array of symbols is a variable or constant
```ruby
  MY_PUBLIC_ALIASES = %i[my_first_alias my_second_alias].freeze
  public(*MY_PUBLIC_ALIASES) # => Style/AccessModifierDeclarations: `public` should not be inlined in method definitions.
```
When the array is not in array literal notation RuboCop suggests array literal notation, that is not being accepted by its sibling cop
```ruby
  MY_PUBLIC_ALIASES = [:my_first_alias, :my_second_alias] # => Style/SymbolArray: Use `%i` or `%I` for an array of symbols.
  public(*MY_PUBLIC_ALIASES) # => Style/AccessModifierDeclarations: `public` should not be inlined in method definitions.
```

## RuboCop version
rubocop -v  => 1.66.1

**Comment by CoderMiguel** [comment]
@koic thank you for the quick response to this issue.  For what it is worth, the PR looks good to me.
